File: Jack.Presentation/Controllers/MVC/CalcadosController.vb
Imports System.Web.Mvc

Namespace Controllers.MVC
    Public Class CalcadosController
        Inherits Controller

        ' GET: Calcados
        Function Index() As ActionResult

            Dim CalcadoBusiness As Business.Calcado = Nothing
            Dim CalcadoRetorno As List(Of Model.Calcado) = Nothing

            Try

                CalcadoBusiness = New Business.Calcado()
                'fake init
                CalcadoRetorno = New List(Of Model.Calcado)
                CalcadoRetorno.Add(New Model.Calcado With {.Codigo = 1, .Numero = 20, .NumeroInicio = 1, .NumeroFim = 6, .Sexo = "M", .MedidaIdade = "M"})
                CalcadoRetorno.Add(New Model.Calcado With {.Codigo = 1, .Numero = 20, .NumeroInicio = 6, .NumeroFim = 12, .Sexo = "M", .MedidaIdade = "M"})
                CalcadoRetorno.Add(New Model.Calcado With {.Codigo = 1, .Numero = 20, .NumeroInicio = 1, .NumeroFim = 2, .Sexo = "M", .MedidaIdade = "A"})
                CalcadoRetorno.Add(New Model.Calcado With {.Codigo = 1, .Numero = 20, .NumeroInicio = 2, .NumeroFim = 3, .Sexo = "M", .MedidaIdade = "A"})
                CalcadoRetorno.Add(New Model.Calcado With {.Codigo = 1, .Numero = 20, .NumeroInicio = 4, .NumeroFim = 5, .Sexo = "M", .MedidaIdade = "A"})
                CalcadoRetorno.Add(New Model.Calcado With {.Codigo = 1, .Numero = 20, .NumeroInicio = 1, .NumeroFim = 6, .Sexo = "F", .MedidaIdade = "M"})
                CalcadoRetorno.Add(New Model.Calcado With {.Codigo = 1, .Numero = 20, .NumeroInicio = 6, .NumeroFim = 12, .Sexo = "F", .MedidaIdade = "M"})
                CalcadoRetorno.Add(New Model.Calcado With {.Codigo = 1, .Numero = 20, .NumeroInicio = 1, .NumeroFim = 2, .Sexo = "F", .MedidaIdade = "A"})
                CalcadoRetorno.Add(New Model.Calcado With {.Codigo = 1, .Numero = 20, .NumeroInicio = 2, .NumeroFim = 3, .Sexo = "F", .MedidaIdade = "A"})
                CalcadoRetorno.Add(New Model.Calcado With {.Codigo = 1, .Numero = 20, .NumeroInicio = 4, .NumeroFim = 5, .Sexo = "F", .MedidaIdade = "A"})
                'fake end
                'CalcadoRetorno = CalcadoBusiness.LoadAll()

            Catch ex As Exception
                CalcadoRetorno = New List(Of Model.Calcado)
            Finally
                CalcadoBusiness = Nothing
            End Try

            Return View(CalcadoRetorno)

        End Function
    End Class
End Namespace

File: Jack.Presentation.Web/Controllers/MVC/CalcadosController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web.Mvc;
using Application = Jack.Application;
using Model = Jack.Model;


namespace Controllers.MVC
{
	public class CalcadosController : Controller
	{

		// GET: Calcados
		public ActionResult Index()
		{

			Application.Calcado CalcadoApplication = null;
			IList<Model.Calcado> CalcadoRetorno = null;


            try
            {
                CalcadoApplication = new Application.Calcado();
                CalcadoRetorno = CalcadoApplication.LoadAll();

            }
            catch (Exception ex)
            {
                CalcadoRetorno = new List<Model.Calcado>();
                throw ex;
            }
            finally
            {
                CalcadoApplication = null;
            }

			return View(CalcadoRetorno);

		}

		public ActionResult Meninos()
		{

			Application.Calcado CalcadoApplication = null;
			IList<Model.Calcado> CalcadoRetorno = null;


            try
            {
                CalcadoApplication = new Application.Calcado();
                CalcadoRetorno = CalcadoApplication.LoadAll().Where(x => x.Sexo == "M").ToList();

            }
            catch (Exception ex)
            {
                CalcadoRetorno = new List<Model.Calcado>();
                throw ex;
            }
            finally
            {
                CalcadoApplication = null;
            }

			return View(CalcadoRetorno);

		}

		public ActionResult Meninas()
		{

			Application.Calcado CalcadoApplication = null;
			IList<Model.Calcado> CalcadoRetorno = null;


            try
            {
                CalcadoApplication = new Application.Calcado();
                CalcadoRetorno = CalcadoApplication.LoadAll().Where(x => x.Sexo == "F").ToList();

            }
            catch (Exception ex)
            {
                CalcadoRetorno = new List<Model.Calcado>();
                throw ex;
            }
            finally
            {
                CalcadoApplication = null;
            }

			return View(CalcadoRetorno);

		}

	}
}
